keep/api/routes/settings.py

```diff
@@ -351,23 +351,118 @@ def _print_debug(self, *args):
             super()._print_debug(*args)
 
 
-@router.get("/apikey")
-def get_api_key(
+@router.post("/apikey", description="Create API key")
+async def create_key(
+    request: Request,
+    authenticated_entity: AuthenticatedEntity = Depends(
+        AuthVerifier(["write:settings"])
+    ),
+    session: Session = Depends(get_session),
+):
+    try:
+        body = await request.json()
+        unique_api_key_id = body['name']
+        role = body['role']
+    except Exception:
+        raise HTTPException(status_code=400, detail="Invalid request body")
+
+    api_key = create_api_key(
+        session=session,
+        tenant_id=authenticated_entity.tenant_id,
+        created_by=authenticated_entity.email,
+        unique_api_key_id=unique_api_key_id,
+        role=role,
+        is_system=False,
+    )
+
+    return api_key
+
+
+@router.get("/apikeys", description="Get API keys")
+def get_keys(
     authenticated_entity: AuthenticatedEntity = Depends(
         AuthVerifier(["read:settings"])
     ),
     session: Session = Depends(get_session),
 ):
-    logger.info("Getting API key")
     tenant_id = authenticated_entity.tenant_id
-    user_name = authenticated_entity.email
-    # get the api key for the CLI
-    api_key = get_or_create_api_key(
+
+    logger.info(f"Getting active API keys for tenant {tenant_id}")
+
+    api_keys = get_api_keys(
+        session=session,
+        tenant_id=tenant_id,
+        email=authenticated_entity.email,
+        role=authenticated_entity.role
+    )
+
+    if api_keys:
+        api_keys = get_api_keys_secret(
+            tenant_id=tenant_id,
+            api_keys=api_keys
+        )
+
+    logger.info(
+        f"Active API keys for tenant {tenant_id} retrieved successfully",
+    )
+
+    return {"apiKeys": api_keys}
+
+
+@router.put("/apikey", description="Update API key secret")
+async def update_api_key(
+    request: Request,
+    authenticated_entity: AuthenticatedEntity = Depends(
+        AuthVerifier(["write:settings"])
+    ),
+    session: Session = Depends(get_session),
+):
+
+    try:
+        body = await request.json()
+        unique_api_key_id = body['apiKeyId']
+    except Exception:
+        raise HTTPException(status_code=400, detail="Invalid request body")
+
+    tenant_id = authenticated_entity.tenant_id
+
+    logger.info(
+        f"Updating API key ({unique_api_key_id}) secret",
+        extra={"tenant_id": tenant_id, "unique_api_key_id": unique_api_key_id},
+    )
+
+    api_key = update_api_key_internal(
```

Comment:
if not api_key shuold return 404 key not found

Reply:
Already returning 404, update_api_key_internal returns a boolean. Look from line 441 to 446